src/socket/tcp.rs

```diff
@@ -3656,33 +3656,25 @@ mod test {
     #[test]
     fn test_buffer_wraparound_tx() {
         let mut s = socket_established();
-        s.tx_buffer = SocketBuffer::new(vec![0; 6]);
-        assert_eq!(s.send_slice(b"abc"), Ok(3));
+        s.tx_buffer = SocketBuffer::new(vec![b'.'; 9]);
+        assert_eq!(s.tx_buffer.enqueue_many(6).len(), 6);
+        assert_eq!(s.tx_buffer.dequeue_many(3).len(), 3);
```

Comment:
Why are you manipulating the buffer directly in this test and not via the socket API? This defeats the purpose of testing the socket.

Reply:
Because this is just preparing the buffer. Anyway, fixed it.